src/main/java/org/cryptacular/util/CertUtil.java

```diff
@@ -454,4 +479,146 @@ public static String authorityKeyId(final X509Certificate cert) throws EncodingE
   {
     return CodecUtil.hex(new ExtensionReader(cert).readAuthorityKeyIdentifier().getKeyIdentifier(), true);
   }
+
+
+  /**
+   * PEM encodes the given certificate with the provided encoding type.
+   *
+   * @param certificate X.509 certificate.
+   * @param encodeType Type of encoding. {@link EncodeType#X509} or {@link EncodeType#PKCS7}
+   *
+   * @return PEM-encoded certificate headed and footer defined by {@link EncodeType} and data wrapped at 64
+   * characters per line.
+   */
+  public static String encodeCert(final X509Certificate certificate, final EncodeType encodeType)
+  {
+    try {
+      return encodeCert(certificate.getEncoded(), encodeType);
+    } catch (CertificateEncodingException e) {
+      throw new RuntimeException("Error getting encoded X.509 certificate data", e);
+    }
+  }
+
+  /**
+   * Converts the given der encoding to PEM encoding.
+   *
+   * @param der DER encoding of certificate with specified encoding.
+   * @param encodeType Encoding type that define the header and footer.
+   *
+   * @return PEM-encoded certificate headed and footer defined by {@link EncodeType} and data wrapped at 64
+   * characters per line.
+   */
+  public static String encodeCert(final byte[] der, final EncodeType encodeType)
+  {
+    final Base64Encoder encoder = new Base64Encoder(64);
+    final ByteBuffer input = ByteBuffer.wrap(der);
+    final CharBuffer output = CharBuffer.allocate(encoder.outputSize(der.length) + 100);
+    output.append("-----BEGIN ").append(encodeType.getType()).append("-----");
+    output.append(UNIX_LINE_SEPARATOR);
```

Comment:
The `Base64Encoder` class use the System line separator but this implementation hard codes unix line separators.